Sources/XCTest/WallClockTimeMetric.swift

```diff
@@ -0,0 +1,86 @@
+    import SwiftFoundation
+#endif
+
+/// This metric uses the system uptime to keep track of how much time passes
+/// between starting and stopping measuring.
+internal final class WallClockTimeMetric: PerformanceMetric {
+    static let name = "org.swift.XCTPerformanceMetric_WallClockTime"
+
+    typealias Measurement = NSTimeInterval
+    private var startTime: NSTimeInterval?
+    var measurements: [Measurement] = []
+
+    func startMeasuring() {
+        startTime = currentTime()
+    }
+
+    func stopMeasuring() {
+        guard let startTime = startTime else { fatalError("Must start measuring before stopping measuring") }
+        let stopTime = currentTime()
+        measurements.append(stopTime-startTime)
+    }
+
+    private let maxRelativeStandardDeviation = 10.0
+    private let standardDeviationNegligibilityThreshold = 0.1
+
+    func calculateResults() -> String {
+        let results = [
+                          String(format: "average: %.3f", measurements.average),
+                          String(format: "relative standard deviation: %.3f%%", measurements.relativeStandardDeviation),
+                          "values: [\(measurements.map({ String(format: "%.6f", $0) }).joined(separator: ", "))]",
+                          "performanceMetricID:\(self.dynamicType.name)",
+                          String(format: "maxPercentRelativeStandardDeviation: %.3f%%", maxRelativeStandardDeviation),
+                          String(format: "maxStandardDeviation: %.3f", standardDeviationNegligibilityThreshold),
+                          ]
+        return "[Time, seconds] \(results.joined(separator: ", "))"
+    }
+
+    func failureMessage() -> String? {
+        // The relative standard deviation of the measurements is \d+.\d{3}% which is higher than the max allowed of \d+.\d{3}%.
+        let relativeStandardDeviation = measurements.relativeStandardDeviation
+        if (relativeStandardDeviation > maxRelativeStandardDeviation &&
+            measurements.standardDeviation > standardDeviationNegligibilityThreshold) {
+            return String(format: "The relative standard deviation of the measurements is %.3f%% which is higher than the max allowed of %.3f%%.", relativeStandardDeviation, maxRelativeStandardDeviation)
+        }
+
+        return nil
+    }
+
+    private func currentTime() -> NSTimeInterval {
+        return NSProcessInfo.processInfo().systemUptime
+    }
+}
+
+
+private extension Collection where Index: IntegerLiteralConvertible, Iterator.Element == WallClockTimeMetric.Measurement {
+    var average: WallClockTimeMetric.Measurement {
+        return self.reduce(0, combine: +) / Double(count.toIntMax())
+    }
+
+    var standardDeviation: WallClockTimeMetric.Measurement {
+        let average = self.average
+        let squaredDifferences = self.map({ pow($0 - average, 2.0) })
+        let variance = squaredDifferences.reduce(0, combine: +) / Double(count.toIntMax()-1)
+        return sqrt(variance)
+    }
```

Comment:
Sorry, that wasn't super clear: this is a nit-pick. Prior to actually having unit tests, I don't feel strongly about whether this should be more generic than it already is.